### ISSUE TITLE
Gpio pins to set at micro-controller startup for Wanhao-D6

### ISSUE THREAD

**Issue by josemabcn** [kushagharahi/wanhao-d6-config#1]
Please, can you share how to build the firmware for the printer?, What microcontroller, processor, etc, and specially the GPIO pins set at micro-controller startup?.

Thanks I'm stopped in this part in order to change the printer from Marlin to Klipper.

Thanks in advance,

Jose

**Comment by kushagharahi** [comment]
Firmware should be compiled for AVR atmega2560, I'm not sure about the other questions. Perhaps you can post a picture?

**Comment by josemabcn** [comment]
Yes, when yo configure to complie the firmware for the printer, the menus shows:

![image](https://user-images.githubusercontent.com/19243737/235320364-8ccf893c-ee55-4992-8919-76804972d8ae.png)

Thanks,

Jose

**Comment by kushagharahi** [comment]
For GPIO pins set at micro-controller startup, I don't think you need to select anything. 

For the other stuff, I think that looks right!

**Comment by josemabcn** [comment]
Thanks for your reply, my last questions is about how to update the firmware, the results form the compile step its a klipper.elf and klipper.elf.hex files. How I can update the firmware?, copy the klipper.elf.hex directly to a SD Car, turn off the printer and turn ON and automatically the printer update the firmware?, I need to rename the klipper.elf.hex file with another name?.

Sorry for the questions...

**Comment by kushagharahi** [comment]
No worries about asking questions! Happy to try and answer them best I can :) 

I don't remember to be honest with you. I remember the PI did all the work when I first set up Klipper (including firmware upload). To be honest, I haven't updated anything since then.

I use mainsailos, perhaps that would be useful? https://docs-os.mainsail.xyz/
